libraries/_helper.rb

```diff
@@ -70,43 +70,32 @@ def delivery_cluster_id
     end
 
     def splunk_server_hostname
-      unless node['delivery-cluster']['splunk']['hostname']
-        node.set['delivery-cluster']['splunk']['hostname'] = "splunk-server-#{delivery_cluster_id}"
-      end
-
-      node['delivery-cluster']['splunk']['hostname']
+      component_hostname('splunk')
     end
 
     def chef_server_hostname
-      unless node['delivery-cluster']['chef-server']['hostname']
-        node.set['delivery-cluster']['chef-server']['hostname'] = "chef-server-#{delivery_cluster_id}"
-      end
-
-      node['delivery-cluster']['chef-server']['hostname']
+      component_hostname('chef-server', 'chef-server')
     end
 
     def delivery_server_hostname
-      unless node['delivery-cluster']['delivery']['hostname']
-        node.set['delivery-cluster']['delivery']['hostname'] = "delivery-server-#{delivery_cluster_id}"
-      end
-
-      node['delivery-cluster']['delivery']['hostname']
+      component_hostname('delivery')
     end
 
     def analytics_server_hostname
-      unless node['delivery-cluster']['analytics']['hostname']
-        node.set['delivery-cluster']['analytics']['hostname'] = "analytics-server-#{delivery_cluster_id}"
-      end
-
-      node['delivery-cluster']['analytics']['hostname']
+      component_hostname('analytics')
     end
 
     def supermarket_server_hostname
-      unless node['delivery-cluster']['supermarket']['hostname']
-        node.set['delivery-cluster']['supermarket']['hostname'] = "supermarket-server-#{delivery_cluster_id}"
+      component_hostname('supermarket')
+    end
+
+    def component_hostname(component, prefix = nil)
+      unless node['delivery-cluster'][component]['hostname']
+        component_prefix = prefix ? prefix : "#{component}-server"
+        node.set['delivery-cluster'][component]['hostname'] = "#{component_prefix}-#{delivery_cluster_id}"
```

Comment:
Oops, yeah I didn't explained that very well, I will elaborate a little bit more about this... :smile: 

That part of code (`node.set`) will only be use when the user do not specify an attribute on the `environment` file for the `hostname` of a component (delivery, chef-server, sumpermarket, etc), we really want to save that value since it is very important for `chef-provisioning` since it is the way we manage the `machine` resources. If we loose that value, we loose the control of the resource and therefore we could potentially provision a new `machine` instead of maintain the one we already provisioned. 

@micgo does that make sense?